pkg/autoupdate/tuf/autoupdate.go

```diff
@@ -67,7 +83,8 @@ func WithUpdateCheckInterval(checkInterval time.Duration) TufAutoupdaterOption {
 	}
 }
 
-func NewTufAutoupdater(metadataUrl, rootDirectory string, metadataHttpClient *http.Client, store types.KVStore, opts ...TufAutoupdaterOption) (*TufAutoupdater, error) {
+func NewTufAutoupdater(metadataUrl, rootDirectory string, updateDirectory string, metadataHttpClient *http.Client,
+	mirrorUrl string, mirrorHttpClient *http.Client, store types.KVStore, osquerier localserver.Querier, opts ...TufAutoupdaterOption) (*TufAutoupdater, error) {
```

Comment:
I don't think we should pull in the `localserver` pkg just for this interface. I'm not sure why `localserver.Querier` was exported in the first placed. We should declare a new interface here. Perhaps if querier is becoming ubiquitious enough, we could create one some where like the `osquery` or `agent` pkg.

Reply:
Yeah, don't pull in `localserver`. Just declare it here.

Counterpoint -- I think it'll move into knapsack, so maybe we just ignore it